Public/Src/Engine/Scheduler/ProcessRunnablePip.cs

```diff
@@ -47,15 +57,27 @@ public sealed class ProcessRunnablePip : RunnablePip
         /// </summary>
         public CacheableProcess CacheableProcess { get; private set; }
 
+        private readonly int m_weightBasedOnHistoricCpuUsage;
+
         internal ProcessRunnablePip(
             LoggingContext phaseLoggingContext,
             PipId pipId,
             int priority,
             Func<RunnablePip, Task> executionFunc,
             IPipExecutionEnvironment environment,
+            ushort cpuUsageInPercents = 0,
             Pip pip = null)
             : base(phaseLoggingContext, pipId, PipType.Process, priority, executionFunc, environment, pip)
         {
+            if (cpuUsageInPercents > 100)
+            {
+                m_weightBasedOnHistoricCpuUsage = (int)Math.Ceiling(cpuUsageInPercents / 100.0);

```

Comment:
>(int)Math.Ceiling(cpuUsageInPercents / 100.0); [](start = 50, length = 46)

Isn't this loosing all resolution? It is now either 0 or 1...

Reply:
ah, understood. Perhaps a copy of the comment on `ProcessorsInPercents` field would be usefull to remind future readers

---
In reply to: [277112078](https://github.com/Microsoft/BuildXL/pull/178#discussion_r277112078) [](ancestors = 277112078)

Reply:
i do not think that we need that resolution at this moment. There are several process pips launching thousands of child processes and using around 600% cpu. The purpose of the changes is to make those pips acquire 6 slots. If there is one using 520% cpu, we do not currently mind that it uses 6 slots.